plugins/woocommerce-admin/client/marketing/overview-multichannel/DiscoverTools/DiscoverTools.tsx

```diff
@@ -45,40 +49,36 @@ const getTabs = ( plugins: Plugin[] ) => {
 	} ) );
 };
 
-const renderPluginCardBodies = ( plugins: Plugin[] ) => {
-	return plugins.map( ( el, idx ) => {
-		return (
-			<Fragment key={ el.product }>
-				<PluginCardBody
-					icon={ <img src={ el.icon } alt={ el.title } /> }
-					name={ el.title }
-					pills={ el.tags.map( ( tag ) => (
-						<Pill key={ tag.slug }>{ tag.name }</Pill>
-					) ) }
-					description={ el.description }
-					button={
-						<Button
-							variant="secondary"
-							href={ getInAppPurchaseUrl( el.url ) }
-							onClick={ () => {
-								recordEvent(
-									'marketing_recommended_extension',
-									{ name: el.title }
-								);
-							} }
-						>
-							{ __( 'Get started', 'woocommerce' ) }
-						</Button>
-					}
-				/>
-				{ idx !== plugins.length - 1 && <CardDivider /> }
-			</Fragment>
-		);
-	} );
-};
-
 export const DiscoverTools = () => {
-	const { isLoading, plugins } = useRecommendedPlugins();
+	const { isLoading, plugins, removeRecommendedPlugin } =
+		useRecommendedPlugins();
+	const [ currentPlugin, setCurrentPlugin ] = useState< string | null >(
+		null
+	);
+	const { installAndActivatePlugins } = useDispatch( PLUGINS_STORE_NAME );
+	const { loadInstalledPluginsAfterActivation } = useInstalledPlugins();
+
+	const installAndActivate = async ( plugin: Plugin ) => {
+		setCurrentPlugin( plugin.product );
+
+		try {
+			recordEvent( 'marketing_recommended_extension', {
+				name: plugin.title,
+			} );
+
+			const response = await installAndActivatePlugins( [
+				plugin.product,
+			] );
+
+			removeRecommendedPlugin( plugin.product );
```

Comment:
Maybe instead of introducing a new method for `removeRecommendedPlugin` we could filter this list for plugins already installed.

I'm hesitant to introduce a new method `removeRecommendedPlugin` that does not persist on page refresh.

Reply:
> I'm hesitant to introduce a new method `removeRecommendedPlugin` that _does not persist on page refresh_.

What do you mean? 🤔 

In the code we have now, after installing a recommended plugin, if we do a page refresh, the list of recommended plugins will not have the installed plugin, due to [filtering in the back-end PHP code](https://github.com/woocommerce/woocommerce/blob/2297787a4f1541a4471c38ccdd44acc936b96bb6/plugins/woocommerce/src/Admin/API/Marketing.php#L117-L119).

`removeRecommendedPlugin` is used to remove the plugin from the recommended list before the page refresh.

Reply:
This has been answered by @joshuatf in https://github.com/woocommerce/woocommerce/pull/35542#issuecomment-1317518587.